src/openstack_cloud/openstack_manager.py

```diff
@@ -497,11 +497,16 @@ def _health_check(
         elapsed_min = (created_at - current_time).total_seconds()
         if server.status == _INSTANCE_STATUS_BUILDING:
             return elapsed_min < CREATE_SERVER_TIMEOUT
-        return OpenstackRunnerManager._ssh_health_check(
-            conn=conn, server_name=server_name, startup=startup
-        )
+        try:
+            return OpenstackRunnerManager._ssh_health_check(
+                conn=conn, server_name=server_name, startup=startup
+            )
+        except _SSHError:
+            logger.warning("Health check failed, unable to SSH into server: %s", server_name)
+            return False
 
     @staticmethod
+    @retry(tries=3, delay=5, max_delay=60, backoff=2, local_logger=logger)
```

Comment:
3 tries with backoff of 2 and delay of 5.
This will result in waiting of 5, 10, 20 seconds.
It might be enough, I think we can try this.